src/mame/machine/gaelco_ds5002fp.cpp

```diff
@@ -0,0 +1,100 @@
+// license:BSD-3-Clause
+// copyright-holders:David Haywood
+#include "emu.h"
+#include "machine/gaelco_ds5002fp.h"
+
+DEFINE_DEVICE_TYPE(GAELCO_DS5002FP,       gaelco_ds5002fp_device,       "gaelco_ds5002fp",      "Gaelco DS5002FP")
+DEFINE_DEVICE_TYPE(GAELCO_DS5002FP_WRALLY, gaelco_ds5002fp_wrally_device, "gaelco_ds5002fp_wr", "Gaelco DS5002FP (World Rally)")
+
+gaelco_ds5002fp_device_base::gaelco_ds5002fp_device_base(machine_config const &mconfig, device_type type, char const *tag, device_t *owner, u32 clock)
+	: device_t(mconfig, type, tag, owner, clock),
+	m_shareram(*this, ":shareram"),
```

Comment:
Please don't hard-code tags at parent or root level.  Provide a macro to configure it.  It creates confusing errors when working on drivers when things do this.